frontend-react/src/hooks/UseValueSets.test.ts

```diff
@@ -0,0 +1,111 @@
+import { rest } from "msw";
+import { renderHook } from "@testing-library/react-hooks";
+import { QueryClient } from "@tanstack/react-query";
+
+import { lookupTableServer } from "../__mocks__/LookupTableMockServer";
+import {
+    LookupTables,
+    ValueSet,
+    lookupTableApi,
+    LookupTable,
+} from "../network/api/LookupTableApi";
+import { QueryWrapper } from "../utils/CustomRenderUtils";
+
+import {
+    useValueSetsTable,
+} from "./UseValueSets";
+
+describe("useValueSetsTable", () => {
+    beforeAll(() => lookupTableServer.listen());
+    afterEach(() => lookupTableServer.resetHandlers());
+    afterAll(() => lookupTableServer.close());
+
+    test("returns expected values when fetching table version", async () => {
+        const { result, waitFor } = renderHook(
+            () => useValueSetsTable<ValueSet>(LookupTables.VALUE_SET),
+            { wrapper: QueryWrapper() }
+        );
+        await waitFor(() => !!result.current.valueSetArray.length);
+        const { name, system, createdAt, createdBy } =
+            result.current.valueSetArray[0];
+        expect(name).toEqual("sender_automation_value_set");
+        expect(createdAt).toEqual("now");
+        expect(createdBy).toEqual("test@example.com");
+        expect(system).toEqual("LOCAL");
+    });
+
+    test("returns expected values when using supplied table version", async () => {
+        const { result, waitFor } = renderHook(
+            () => useValueSetsTable<ValueSet>(LookupTables.VALUE_SET, 3),
+            { wrapper: QueryWrapper() }
+        );
+        await waitFor(() => !!result.current.valueSetArray.length);
+        const { name, system, createdAt, createdBy } =
+            result.current.valueSetArray[0];
+        expect(name).toEqual("sender_automation_value_set");
+        expect(createdAt).toEqual(undefined); // not expecting this data to be present in this case
+        expect(createdBy).toEqual(undefined); // not expecting this data to be present in this case
+        expect(system).toEqual("LOCAL");
+    });
+
+    test("returns error when the passed table name doesn't exist in returned list of tables", async () => {
+        const { result, waitFor } = renderHook(
+            () => useValueSetsTable(LookupTables.VALUE_SET_ROW),
+            { wrapper: QueryWrapper() }
+        );
+        await waitFor(() => !!result.current.error);
+        expect(result.current.error.message).toEqual(
+            `Table 'sender_automation_value_set_row' was not found!`
+        );
+    });
+
+    test("returns error when table list call errors", async () => {
```

Comment:
Because of how long the console errors from react-query are, can we do a mock for `console.error` in this file?

Reply:
yeah. Maybe it's worth bringing in that tech debt ticket at this point?

Reply:
Which ticket in particular? Couldn't find it but I know we've talked about maybe wanting to use the errors somehow in tests

Reply:
this is the ticket https://app.zenhub.com/workspaces/experience-607d9d5e68b95200150fec37/issues/cdcgov/prime-reportstream/6289. If I'm adding `--silent` here I guess that would close the ticket. I'm a fan of that approach - if you need logs for debugging you can just remove the flag. I could do something like this though
```
beforeAll(() => {
  jest.spyOn(console, 'log').mockImplementation(() => {});
  jest.spyOn(console, 'error').mockImplementation(() => {});
  jest.spyOn(console, 'warn').mockImplementation(() => {});
});
```